src/storage/accessors/FileDataAccessor.ts

```diff
@@ -297,10 +297,16 @@ export class FileDataAccessor implements DataAccessor {
         continue;
       }
 
-      // Generate metadata of this specific child
+      // Generate metadata of this specific child as described in
+      // https://solidproject.org/TR/2021/protocol-20211217#contained-resource-metadata
       const metadata = new RepresentationMetadata(childLink.identifier);
       addResourceMetadata(metadata, childStats.isDirectory());
       this.addPosixMetadata(metadata, childStats);
+      // Containers will not have a content-type
+      if (childLink.contentType) {
+        metadata.add(RDF.terms.type, toNamedTerm(`${IANA.namespace}${childLink.contentType}#Resource`));
```

Comment:
We'll need to sanity-check `childLink.contentType` unfortunately. Unless this is done elsewhere—but still. We don't want to be tricked into writing weird things.

Reply:
What exactly is the potential issue that you see here? It will always just be a named node in a triple, so converters should have no issue with it to convert it to valid turtle (or other format).

Reply:
`childLink.contentType = 'a very weird content type'` resulting in an invalid RDF term, hence invalid RDF, hence parser failure.

Reply:
If we write invalid RDF I would see that as an issue with our serializer since that is going to convert this NamedNode to turtle and is responsible for generating valid RDF. But I'll just add a `parseContentType` wrapped by a try/catch here to make sure the request doesn't fail if there is a child with an invalid type.